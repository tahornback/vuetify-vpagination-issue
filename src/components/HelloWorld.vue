<template>
  <p arialabel="paragraph using arialabel">arialabel in Vue</p>
  <p ariaLabel="paragraph using ariaLabel">ariaLabel in Vue</p>
  <p aria-label="paragraph using aria-label">aria-label in Vue</p>
</template>

<script setup>
  //
</script>
