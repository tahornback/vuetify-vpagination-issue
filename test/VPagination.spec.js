import {render,screen} from "@testing-library/vue";
import {createVuetify} from "vuetify";
import {VPagination} from "vuetify/components";
import {components, directives} from "vuetify/dist/vuetify";
import '@testing-library/jest-dom'

const vuetify = createVuetify({
    components,
    directives,
})

global.ResizeObserver = require('resize-observer-polyfill')

describe('VPagination Current', () => {
    it('renders buttons with arialabel', () => {
        const wrapper = render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
        screen.getAllByRole('button').forEach(b => {
            expect(b).toHaveAttribute('arialabel')
        })
    })

    it('renders non current-page buttons with ariadisabled', () => {
        const wrapper = render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
        screen.getAllByRole('button').forEach(b => {
            if(!b.hasAttribute('ariacurrent'))
                expect(b).toHaveAttribute('ariadisabled')
        })
    })

    it('renders current page button with ariacurrent', () => {
        const wrapper = render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
        expect(screen.getByText('1').parentElement).toHaveAttribute('ariacurrent')
    })
})

describe.skip('VPagination Expected', () => {
    it('renders buttons with aria-label', () => {
        render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        screen.getAllByRole('button').forEach(b => {
            expect(b).toHaveAttribute('aria-label')
        })

        screen.getByLabelText('Previous page') // This does not work in Vuetify 3.4.7 since `arialabel` isn't a proper label
    })

    it('renders non current-page buttons with aria-disabled', () => {
        render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        screen.getAllByRole('button').forEach(b => {
            if(!b.hasAttribute('aria-current'))
                expect(b).toHaveAttribute('aria-disabled')
        })
    })

    it('renders current page button with aria-current', () => {
        const wrapper = render(VPagination, {
            global: {
                plugins: [vuetify]
            }
        })

        expect(screen.getByLabelText('Page 1, Current page').parentElement).toHaveAttribute('aria-current') // This does not work in Vuetify 3.4.7 since `arialabel` isn't a proper label
    })
})
