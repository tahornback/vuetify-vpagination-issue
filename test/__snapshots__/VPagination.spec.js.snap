// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VPagination Current > renders buttons with arialabel 1`] = `
"<nav class="v-pagination v-theme--light" role="navigation" aria-label="Pagination Navigation" data-test="v-pagination-root">
  <ul class="v-pagination__list">
    <!---->
    <li class="v-pagination__prev" data-test="v-pagination-prev"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Previous page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__item v-pagination__item--is-active" data-test="v-pagination-item"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" ellipsis="false" ariacurrent="true" arialabel="Page 1, Current page"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator="">1</span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__next" data-test="v-pagination-next"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Next page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <!---->
  </ul>
</nav>"
`;

exports[`VPagination Current > renders current page button with ariacurrent 1`] = `
"<nav class="v-pagination v-theme--light" role="navigation" aria-label="Pagination Navigation" data-test="v-pagination-root">
  <ul class="v-pagination__list">
    <!---->
    <li class="v-pagination__prev" data-test="v-pagination-prev"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Previous page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__item v-pagination__item--is-active" data-test="v-pagination-item"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" ellipsis="false" ariacurrent="true" arialabel="Page 1, Current page"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator="">1</span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__next" data-test="v-pagination-next"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Next page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <!---->
  </ul>
</nav>"
`;

exports[`VPagination Current > renders non current-page buttons with ariadisabled 1`] = `
"<nav class="v-pagination v-theme--light" role="navigation" aria-label="Pagination Navigation" data-test="v-pagination-root">
  <ul class="v-pagination__list">
    <!---->
    <li class="v-pagination__prev" data-test="v-pagination-prev"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Previous page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__item v-pagination__item--is-active" data-test="v-pagination-item"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" ellipsis="false" ariacurrent="true" arialabel="Page 1, Current page"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator="">1</span>
        <!---->
        <!---->
      </button></li>
    <li class="v-pagination__next" data-test="v-pagination-next"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text" disabled="" arialabel="Next page" ariadisabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></li>
    <!---->
  </ul>
</nav>"
`;
